server/src/test/java/com/objectcomputing/checkins/endtoend/pages/HomePage.java

```diff
@@ -1,72 +1,79 @@
 import org.openqa.selenium.WebElement;
 import org.openqa.selenium.support.FindBy;
 
+import java.io.File;
+import java.net.URL;
+
 import static org.junit.jupiter.api.Assertions.assertEquals;
 
 /**
  * @author tajr
- * 
+ *
  * This class will store all the locators and methods of home page.
  *
  */
 public class HomePage {
-	
-	WebDriver driver;
-	
-	public HomePage(WebDriver ldriver)
-	{
-		this.driver = ldriver;
-	}
-	
-	@FindBy(xpath="/html[1]/body[1]/div[1]/form[1]/fieldset[1]/label[1]") 
-	WebElement chooseafile_btn;
-	
-	@FindBy(id="file")
-	WebElement filepath;
-	
-	@FindBy(name="submit")
-	WebElement upload_btn;
 
-	@FindBy(className="success")
-	WebElement success_msg;
-	
-	@FindBy(className="error")
-	WebElement error_msg;
-	
-	
+    WebDriver driver;
+    @FindBy(xpath = "/html[1]/body[1]/div[1]/form[1]/fieldset[1]/label[1]")
+    WebElement chooseAFileBtn;
+    @FindBy(id = "file")
+    WebElement filepath;
+    @FindBy(name = "submit")
+    WebElement uploadBtn;
+    @FindBy(className = "success")
+    WebElement successMsg;
+    @FindBy(className = "error")
+    WebElement errorMsg;
+
+    public HomePage(WebDriver ldriver) {
+        this.driver = ldriver;
+    }
+
+    public void uploadFileSuccessMsg() throws InterruptedException {
+        String successMsg = this.successMsg.getText();
+        System.out.println("Upload File Msg: " + successMsg);
+        Thread.sleep(1000);
+        assertEquals(this.successMsg.getText(), "THE FILE APPLICATION.YML WAS UPLOADED");
+        Thread.sleep(5000);
+    }
+
+
 	public void uploadFile() throws InterruptedException
 	{
-		// TODO needs to be fixed
-		filepath.sendKeys("/Users/tajr/Desktop/File.rtf");
+		File file = getFileFromResources("application.yml");
+		filepath.sendKeys(file.getPath());
 		Thread.sleep(5000);
-		upload_btn.click();
+		uploadBtn.click();
 		Thread.sleep(5000);
 	}
-	
-	public void uploadFileSuccessMsg() throws InterruptedException
-	{
-		String successMsg = success_msg.getText();
-		System.out.println("Upload File Msg: " +successMsg);
-		Thread.sleep(1000);	
-		assertEquals(success_msg.getText(),"THE FILE FILE.RTF WAS UPLOADED");
-		Thread.sleep(5000);
-	}
-		
-	public void uploadFileErrorMsg() throws InterruptedException
-	{
-		upload_btn.click();
-		Thread.sleep(5000);
-		String errorMsg = error_msg.getText();
-		System.out.println("Upload File Msg: " +errorMsg);
-		Thread.sleep(1000);	
-		assertEquals(error_msg.getText(),"Please select a file before uploading.");
-		Thread.sleep(5000);
+
+	private File getFileFromResources(String fileName) {
+
+		ClassLoader classLoader = getClass().getClassLoader();
+
+		URL resource = classLoader.getResource(fileName);
+		if (resource == null) {
+			throw new IllegalArgumentException("file is not found!");
+		} else {
+			return new File(resource.getFile());
+		}
+
 	}
+    public void uploadFileErrorMsg() throws InterruptedException {
+        uploadBtn.click();
+        Thread.sleep(5000);
```

Comment:
Since the code define an implicit wait `driver.manage().timeouts().implicitlyWait(30, TimeUnit.SECONDS);`, could these `Thread.sleep()`'s be replaced with a wait for element to be visible or something similar?

Reply:
Probably...I think we have someone who is going to be diving into these tests more deeply. I am going to go ahead and merge this one as is and create an issue to remove `sleep` calls where possible.